test/denormalized_has_many_test.rb

```diff
@@ -12,7 +12,23 @@ def setup
     @record.reload
   end
 
+  def test_uncached_record_from_the_db_will_use_normal_association_as_ary
+    # Test configuration where fetch association returns an array
+    IdentityCache.allow_fetch_to_return_array do    
+      assert_equal IdentityCache.fetch_returns_relation, false
+      expected = @record.associated_records.to_a
+      record_from_db = Item.find(@record.id)
+      assert_queries(1) do
+        assert_memcache_operations(0) do
```

Comment:
I think you are trying to test too many things here.  Try to keep the test cases more focused.  I don't think you need to test the number of queries in this case.